app-modules/authorization/tests/Feature/Filament/Resources/User/RestoreUserTest.php

```diff
@@ -0,0 +1,157 @@
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Models\User;
+
+use function Tests\asSuperAdmin;
+use function Pest\Laravel\actingAs;
+use function Pest\Livewire\livewire;
+
+use Filament\Tables\Actions\RestoreAction;
+use Filament\Tables\Filters\TrashedFilter;
+use App\Filament\Resources\UserResource\Pages\EditUser;
+use App\Filament\Resources\UserResource\Pages\ListUsers;
+use App\Filament\Resources\UserResource\Pages\CreateUser;
+
+it('show trashed filter only if user has user restore permission', function () {
+    $user = User::factory()->create();
+
+    $user->givePermissionTo('user.view-any', 'user.*.view', 'user.*.restore');
+    actingAs($user);
+
+    livewire(ListUsers::class)
+        ->assertTableFilterExists(TrashedFilter::class);
+});
+
+it('do not show soft deleted users when filter is not selected', function () {
+    asSuperAdmin();
+    $users = User::factory()->count(3)->create();
+
+    // Soft-delete one of the users
+    $softDeletedUser = $users->first();
+    $softDeletedUser->delete();
+    $nonDeletedUserRecords = User::get();
+    $softDeletedUserRecords = User::onlyTrashed()->get();
+
+    livewire(ListUsers::class)
+        ->removeTableFilters()
+        ->assertCanSeeTableRecords($nonDeletedUserRecords)
+        ->assertCanNotSeeTableRecords($softDeletedUserRecords);
+});
+
+it('can see soft deleted, non soft deleted records and status column only if the filter is on', function () {
+    asSuperAdmin();
+    $users = User::factory()->count(3)->create();
+
+    // Soft-delete one of the users
+    $softDeletedUser = $users->first();
+    $softDeletedUser->delete();
+
+    livewire(ListUsers::class)
+        ->filterTable(TrashedFilter::class)
+        ->assertTableColumnExists('deleted_at')
+        ->assertCanSeeTableRecords($users);
+});
+
+it('Show restore action only if user has permission to restore user', function () {
+    $user = User::factory()->create();
+
+    $user->givePermissionTo('user.view-any', 'user.*.view', 'user.*.restore');
+    actingAs($user);
+
+    $users = User::factory()->count(3)->create();
+
+    // Soft-delete one of the users
+    $softDeletedUser = $users->first();
+    $softDeletedUser->delete();
+    $softDeletedUserRecords = User::onlyTrashed()->get();
+
+    livewire(ListUsers::class)
+        ->filterTable(TrashedFilter::class, 0)
+        ->assertCanSeeTableRecords($softDeletedUserRecords)
+        ->assertTableActionExists(RestoreAction::class);
```

Comment:
You will want to check if the action exists but it not visible before giving the `user.*.restore`, then give the user the permission and then test that it is visible